models/user.go

```diff
@@ -223,7 +223,7 @@ func (u *User) CanCreateRepo() bool {
 
 // CanCreateOrganization returns true if user can create organisation.
 func (u *User) CanCreateOrganization() bool {
-	return u.IsAdmin || u.AllowCreateOrganization
+	return u.IsAdmin || (u.AllowCreateOrganization && !setting.Admin.DisableRegularOrgCreation)
```

Comment:
default value as `true` makes sense for me. Admin need to update all user setting if you set `DisableRegularOrgCreation` as `false` again.